### PR TITLE
Use off-the-shelf React Templates for UI

### LINKED ISSUE #86: Improve chat line rendering
## Problem

`Card` is used for chat line rendering:

https://github.com/status-im/js-waku/blob/72c9bef3d2c171e16689e4c9d66364f33d110ad6/web-chat/src/ChatList.tsx#L49

This is not great as the messages' height are just too big. It's using more vertical space than needed

Edit: We will dog food the app to replace the current \#waku discord channel. The channel can get very busy, especially overnight. Hence, the theme needs to be a la discord/slack/element.

## Definition of Done

Have a rendering similar to what is currently done and most modern chat apps:

- login and date on top left corner
- login in small case
- date in smaller case than login
- message in standard/normal case left aligned

### PR DESCRIPTION
**Pull request by littlealex003**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactor

- **What is the current behavior?** (You can also link to an open issue here)
Custom React Material UI Implementation

- **What is the new behavior (if this is a feature change)?**
Off the shelf React Material Chat UI Implementation

### CODE REVIEW

File: web-chat/src/ChatList.tsx [hunk 1]
```diff
@@ -1,73 +1,38 @@
...
 
-  const listItems = messages.map((message) => (
-    <ListItem key={message.timestamp.toString()}>
-      <ListItemText primary={<Message message={message} />} />
-    </ListItem>
+  const listItems = messages.map((currentmessage) => (
+    <Message
+      key={currentmessage.timestamp.toString()}
+      authorName={currentmessage.nick}
+      date={currentmessage.timestamp.toLocaleString([], {
```
Comment (D4nte) [inline web-chat/src/ChatList.tsx:17]: Maybe best if we extract the formatting of the date in a dedicated function to declutter.

File: web-chat/src/MessageInput.tsx [hunk 2]
```diff
@@ -1,7 +1,15 @@
-import React, { ChangeEvent, KeyboardEvent, useState } from 'react';
-import { Button, Grid, TextField } from '@material-ui/core';
+import { ChangeEvent, KeyboardEvent, useState } from 'react';
 import { useWaku } from './WakuContext';
 
+import {
```
Comment (D4nte) [inline web-chat/src/MessageInput.tsx:4]: Did the linter add this newline?
```suggestion
import {
```

File: web-chat/src/custom.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+declare module '@livechat/ui-kit';
```
Comment (D4nte) [inline web-chat/src/custom.d.ts:1]: To keep it consistent with js-waku, name the file `types.d.ts` and add it to a `types` folder.

You probably also need to update the `web-chat/tsconfig.json` file with

```
 "typeRoots": ["node_modules/@types", "src/types"]
```

like `./tsconfig.json`.

**Review by D4nte — COMMENTED** [review]
I must say it looks good and it is an improvement on the current UX so I'd be keen to merge even if it is not the slack/discord feel.

Can you make it so that messages from the same sender are automatically grouped? This should help make it more condensed.

**Comment by D4nte** [comment]
`Resolves #86` needs to be in the description, not the title to work. See https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

**Comment by D4nte** [comment]
Can we get rid of material-ui with this PR?

**Comment by littlealex003** [comment]
Sure, we can improve the look and feel and customize it and group them,
honestly I didn't realize we were planning on actually using it and was
trying to leave it with as little UI code clutter as possible and looking
as much as it did before by just replacing the custom stuff with the off
the shelf components.

I can group the messages by sender and in general make it a lot smaller,
and get rid of the bubbles; I misunderstood the goal to be more about
making a demo of how to integrate it than a demo we would use.

I can refactor the date function; I did try to remember to run the linter
before every commit; I'll refactor the JS->TS integration code, I was just
doing something to make it work for the demo, I'll try to match what you
have.

On Thu, Apr 29, 2021 at 2:55 AM Franck Royer ***@***.***>
wrote:

> *@D4nte* commented on this pull request.
>
> I must say it looks good and it is an improvement on the current UX so I'd
> be keen to merge even if it is not the slack/discord feel.
>
> Can you make it so that messages from the same sender are automatically
> grouped? This should help make it more condensed.
>
------------------------------

**Comment by littlealex003** [comment]
Ah ok, sorry I'm more used to using JIRA than the built in Github issue
system (I'm also more used to using Jenkins than their built in CI)

Honestly I've been using Google cloud git and Gitlab a lot for the last
years.

**Comment by littlealex003** [comment]
Yes you are right, there are no more dependencies on Material UI, I'll
remove it

**Comment by D4nte** [comment]
Thanks for the replies!

Feel free to add a pre-commit hook to lint the code automatically.
Yes, removing bubble and grouping by sender would be great.

Indeed, I hadn't clarify the fact that we were also dogfooding the app (and it took me some times to remember why I wanted it condensed).

Again, the look is much better so I am excited to get this merged with those small tweaks.

**Comment by littlealex003** [comment]
No worries! We are on almost perfectly opposite times...you had plenty of
time to review everything while I slept. First thing in the morning with my
coffee is a good time for me to go over all this stuff.

**Comment by littlealex003** [comment]
I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.

Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?

File: web-chat/src/ChatList.tsx (hunk 1 above)
Comment (littlealex003) [reply web-chat/src/ChatList.tsx]: Done (please review)

File: web-chat/src/MessageInput.tsx (hunk 2 above)
Comment (littlealex003) [reply web-chat/src/MessageInput.tsx:4]: I'm guessing so, but it's possible VSCode did; I've been running "npm run fix" on the web-chat directory before I commit, but it should be automated with a pre-commit hook as you mentioned.

File: web-chat/src/custom.d.ts (hunk 3 above)
Comment (littlealex003) [reply web-chat/src/custom.d.ts]: Ok should be done now, sorry :)

**Comment by D4nte** [comment]
> I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.
> 
> Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?

It is fine if you want to get this merged first and then work on getting it more condensed in a second PR.
If that's what you prefer to do then yes, please remove `resolves #86`.

**Comment by littlealex003** [comment]
> > I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.
> > Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?
> 
> It is fine if you want to get this merged first and then work on getting it more condensed in a second PR.
> If that's what you prefer to do then yes, please remove `resolves #86`.

I removed it and squashed everything into a single commit. I think the linter may have reformatted the tsconfig; I can revert that if you prefer, let me know.

I like to do smaller more atomic commits while I'm working but not too hard to squash them all into a single commit at the end.

**Comment by D4nte** [comment]
Looks good, I'll take over

**Review by D4nte — APPROVED** [review]

**Comment by D4nte** [comment]
bors merge

**Review by D4nte — APPROVED** [review]

**Comment by D4nte** [comment]
bors merge

**Pull request merged**